### ISSUE TITLE
Sometimes board fails to boot

### ISSUE THREAD

**Issue by hallor** [hallor/u-boot#9]
version: 2b9aaa5b02d5e04a57e64cae0bb5e4b97589f69e
image: http://people.linaro.org/~nicolas.dechesne/u-boot-hallor.img
compiler: linaro fcc 4.9 2014.09

Sometimes fails to boot, sometimes resets at prompt, sometimes works.

Boot config setting: 0100 (try SD boot)

HDMI, no SD

**Comment by hallor** [comment]
Issue reproduced on my setup, it doesn't reproduce if I compile my own u-boot from 2b9aaa5b02d5e04a57e64cae0bb5e4b97589f69e version.

**Comment by hallor** [comment]
It happens only on specific Linaro toolchain; doesn't happen on kernel.org 4.9 toolchain

**Comment by afaerber** [comment]
I encountered it with gcc-linaro-4.9-2015.02-3-x86_64_aarch64-elf.

gcc-linaro-5.1-2015.08-x86_64_aarch64-linux-gnu works okay.

**Comment by hallor** [comment]
@afaerber Thanks for the testing; Did you ran image from this issue, or build it yourself?

**Comment by afaerber** [comment]
No, it was a follow-up to my u-boot list message and the v2 branch mentioned there.
